WooCommerce/Classes/Tools/VersionHelpers.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+
+/// Helpers for working with versions (e.g. comparing two version strings)
+///
+final class VersionHelpers {
+    private init() {}
+
+    /// Compares two strings as versions using the same approach as PHP `version_compare`.
+    /// https://www.php.net/manual/en/function.version-compare.php
+    ///
+    /// Returns `orderedAscending` if the lhs version is older than the rhs
+    /// Returns `orderedSame` if the lhs version is the same as the rhs
+    /// Returns `orderedDescending` if the lhs version is newer than the rhs
+    ///
+    static func compare(_ lhs: String, _ rhs: String) -> ComparisonResult {
+
+        /// Replace _ - and + with a .
+        ///
+        func replaceUnderscoreDashAndPlusWithDot(_ string: String) -> String {
+            string.replacingOccurrences(of: "([_\\-+]+)", with: ".", options: .regularExpression)
+        }
+
+        /// Insert a . before and after any non number
+        ///
+        func insertDotsBeforeAndAfterAnyNonNumber(_ string: String) -> String {
+            string.replacingOccurrences(of: "([^0-9.]+)", with: ".$1.", options: .regularExpression)
+        }
+
+        /// Score and compare two string components
+        ///
+        func compareStringComponents(_ lhs: String, _ rhs: String) -> ComparisonResult {
+            /// Score each component
+            let lhsScore = VersionComponentScore(from: lhs)
+            let rhsScore = VersionComponentScore(from: rhs)
+
+            if lhsScore < rhsScore {
+                return .orderedAscending
+            }
+
+            if lhsScore > rhsScore {
+                return .orderedDescending
+            }
+
+            if lhsScore == .number && rhsScore == .number {
+                let lhsAsNumber = NSNumber(value: Int(lhs) ?? 0)
+                let rhsAsNumber = NSNumber(value: Int(rhs) ?? 0)
+
+                let comparisonResult = lhsAsNumber.compare(rhsAsNumber)
+                if comparisonResult != .orderedSame {
+                    return comparisonResult
+                }
+            }
+
+            return .orderedSame
+        }
+
+        /// Process the given string into version components
+        ///
+        func versionComponents(of string: String) -> [String] {
+            var stringToComponentize = replaceUnderscoreDashAndPlusWithDot(string)
+            stringToComponentize = insertDotsBeforeAndAfterAnyNonNumber(stringToComponentize)
+            return stringToComponentize.components(separatedBy: ".")
+        }
+
+        let leftComponents = versionComponents(of: lhs)
+        let rightComponents = versionComponents(of: rhs)
+
+        let maxComponents = max(leftComponents.count, rightComponents.count)
+
+        for index in 0..<maxComponents {
+            /// Treat missing components (e.g. 1.2 being compared to 1.1.3 as "0", i.e. 1.2.0
```

Comment:
super nit:
```suggestion
            /// Treat missing components (e.g. 1.2 being compared to 1.1.3 as "0", i.e. 1.2.0)
```

Reply:
Fixed as part of https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/2eb5cb0402c9bef48cd7b74fe3d81bbe031f3c44